adagrams/game.py

```diff
@@ -1,11 +1,100 @@
+import random 
+import copy
 def draw_letters():
-    pass
+    LETTER_POOL = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+#make list to return for "hand"
+#draw 10 random letters from pool
+#reflect letter distribution thats in dict
+    hand = []
+    for letter, amount in LETTER_POOL.items():
+        for num in range(amount):
+            hand.append(letter)
+            random.shuffle(hand)
+    return hand[:10]
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_copy = copy.deepcopy(letter_bank)
+    for char in word.upper():
+        if char not in letter_bank_copy:
+            return False
+        elif char in letter_bank_copy:
+            letter_bank_copy.remove(char)
+    return True 
 
 def score_word(word):
-    pass
+    # passes in a string
+    # returns an int representing total score
+    # if len of word 7-10 than +8 points
+    letter_value = {
+    ('A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'): 1,
+    ('D', 'G'): 2,
+    ('B', 'C', 'M', 'P'): 3,
+    ('F', 'H', 'V', 'W', 'Y'): 4,
+    ('K',): 5,
+    ('J', 'X'): 8,
+    ('Q', 'Z'): 10
+    }
+    score = []
+    # iterate through dictionary
+    for letter, point in letter_value.items():
+        for char in word:
+            # if char in word also in keys of dict, add the associated point
+            if char.upper() in letter:
+                score.append(point)
+    #if the length of the word is between 7-10, add 8 pts
+    if len(word) in range(7, 11):
+        score.append(8)
+            
+    return sum(score)
 
```

Comment:
Interesting approach to this function 😁 I love how you created a list of the scores 👍🏾 you could also set the score to `0` and use compound assignment to update the score